### PR TITLE
feat(threshold): Add threshold arguments for the CLI

### LINKED ISSUE #11: Thresholds
Right now the mutation score color thresholds are fixed at 60 and 80.

It would be nice to let users pass their own preferred thresholds. Also an extra threshold could be added for returning a non zero exit code.

Param suggestion: `--thresholds 60 80 90` for:

- `< 60` return exit code 1
- `> 60` mutation score is red
- `> 80` mutation score is yellow
- `> 90` mutation score is green

### PR DESCRIPTION
**Pull request by Stukongeluk**
fixes #11 

I saw that there was another issue related to this one (#121), but that won't be fixed in this PR. However, there is a little setup for it to make it easier to fix.

A note with this PR:
- I've changed the test in `ConsoleReporterReporterTest.cs`, but I'm not sure if this is the right way to do this.  My knowledge about testing in C# is very <b> VERY </b> basic.

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs [hunk 1]
```diff
@@ -54,5 +65,28 @@ private LogEventLevel ValidateLogLevel(string levelText)
...
+                   throw new ValidationException("The threshold can never be > 100");
+               } else if (threshold < 0)
+               {
+                   throw new ValidationException("The threshold can not be < 0");
+               }
+            }
+
+            if (thresholdList[0] >= thresholdList[1] || 
+                thresholdList[1] >= thresholdList[2] || 
+                thresholdList[0] >= thresholdList[2])
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs]: When `0 >= 1` and `1 >= 2` , 0 is always >= 2. So the last conditional is not necessary.

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs [hunk 2]
```diff
@@ -17,13 +17,24 @@ public class StrykerOptions
 
         public int AdditionalTimeoutMS { get; }
 
-        public StrykerOptions(string basePath, string reporter, string projectUnderTestNameFilter, int additionalTimeoutMS, string logLevel, bool logToFile)
+        public int ThresholdBreak { get; }
```
Comment (simondel) [inline src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs]: I'd prefer to store the Threshold as an object on the `StrykerOptions` class with separate properties. I would also expect this to be defined as a JSON object if you use a config file.

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs [hunk 3]
```diff
@@ -17,13 +17,24 @@ public class StrykerOptions
...
+
+        public StrykerOptions(string basePath, string reporter, string projectUnderTestNameFilter, int additionalTimeoutMS, string logLevel, bool logToFile, 
+        int thresholdBreak, int thresholdLow, int thresholdHigh)
         {
             BasePath = basePath;
             Reporter = ValidateReporter(reporter);
             ProjectUnderTestNameFilter = projectUnderTestNameFilter;
             AdditionalTimeoutMS = additionalTimeoutMS;
             LogOptions = new LogOptions(ValidateLogLevel(logLevel), logToFile);
+            int[] ThresholdList = ValidateThresholds(new int[3]{thresholdBreak, thresholdLow, thresholdHigh});
```
Comment (simondel) [inline src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs]: I don't like the idea that you're moving the properties into an array and then retreiving them from the array again. You don't know if the `ValidateThresholds` method hasn't changed the order of the Thresholds. 

It doesn't do anything with the list so we don't need to return it. If we do have to return it, we could consider using a tuple.

**Comment by Stukongeluk** [comment]
Thanks for the feedback @simondel & @richardwerkman ! The next commit will fix all these and also fix the conflicts ;)

**Comment by Stukongeluk** [comment]
Merge conflicts fixed

File: src/Stryker.CLI/README.md [hunk 4]
```diff
@@ -67,9 +67,21 @@ Example:
         "timeout-ms":2000,
         "log-file":true,
         "project-file":"ExampleProject.csproj",
-		"max-concurrent-test-runners": 4
+		"max-concurrent-test-runners": 4,
```
Comment (richardwerkman) [inline src/Stryker.CLI/README.md]: Tab

**Comment by Stukongeluk** [comment]
I didn't add the `threshold-break` case to the readme.md, because it still isn't implemented to let stryker exit. I think the one who picks up #121 should add this to the readme, because after that PR, it should actually work like documented.

**Pull request merged**